File: 0x00 基本算法/0x07 贪心/AcWing1055.cpp
#include <bits/stdc++.h>
using namespace std;
const int N = 1e5 + 10;
int n, p[N];

// 缩小求解空间：可将跨越天数>1的交易拆分成若干个跨越天数为1的交易；n-1个交易互相独立
int main() {
  cin >> n;
  for (int i = 0; i < n; i ++ ) cin >> p[i];

  int res = 0;
  for (int i = 0; i < n - 1; i ++ ) res += max(0, p[i + 1] - p[i]);

  cout << res;
  return 0;
}

File: 0x00 基本算法/0x07 贪心/AcWing1055.py
import sys

def main():
    input_data = sys.stdin.read().split()
    if not input_data:
        return
        
    n = int(input_data[0])
    p = [int(x) for x in input_data[1:n + 1]]
    
    res = sum(max(0, p[i + 1] - p[i]) for i in range(n - 1))
    print(res)

if __name__ == '__main__':
    main()